MantidQt/API/src/WorkspaceObserver.cpp

```diff
@@ -58,6 +62,14 @@ namespace MantidQt
       m_rename_observed(false), m_clr_observed(false)
     {
       m_proxy = new ObserverCallback(this);
+
+	  static bool registered_addtional_types = false;
+	  if (!registered_addtional_types)
+	  {
+		  registered_addtional_types = true;
```

Comment:
Is this just to ensure that we only call `registered_additional_types` once? If so `std::call_once` would be a better bet. However. This shouldn't live in WorkspaceObserver, constructor as it requires that that constructor is called at least once for this to work properly. The solution should be to put an `%InitialisationCode
    code
%End` block into the mantidqt.sip file at the beginning with `qRegisterMetaType<Mantid::API::Workspace_sptr>()` and `qRegisterMetaType<std::string>()` entered at that point. That way it will happen once for the module.